rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,505 @@
+- Help users avoid and correct mistakes.
+
+**Robust**
+
+- Maximize compatibility with current and future user tools.
+
+# Motivation
+- Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+## Why are we doing this? 
+
+Without proper accessibility, certain functionality used in web sites is not available to some users with disabilities, especially people who rely on screen readers and people who cannot use a mouse. 
+
+Making our products accessible is important for two reasons.  
+- it's the right thing to do. We believe that everyone deserves to be able to use our products, regardless of their abilities.
+- It'll help build trust with our customers. By making our products accessible, we can expand our potential customer base and make our products more appealing to a wider range of people. 
+
+> [The Business Case Study for Accessibility](https://www.w3.org/WAI/business-case/)
+
+**Making customers happy & respecting human rights:**  
+
+> In 2020, A 29 year old banker who is visually impaired opened petitions to make [zomato](https://www.change.org/p/zomato-make-zomato-accessible-with-screen-readers-for-visually-challenged-people) & [swiggy](https://www.change.org/p/swiggy-make-swiggy-accessible-with-screen-readers-in-android-for-visually-challenged-people) accessible because they couldn’t order food from it due to their inaccessible apps.  
+> Over **15,000 & 27,000 people had signed the petition.**  
+> Later zomato, swiggy listened to the people and [improved their accessibility.](https://theprint.in/tech/why-the-visually-challenged-are-signing-petitions-addressed-to-apps-such-as-zomato-swiggy/507282/) 
+
+**Accessibility laws & policies** 
+
+While in India, there's no mandatory law to building accessible web apps, 
+There's still [policies/laws](https://www.w3.org/WAI/policies/india/) which are passed to ensure rights of disabled people remains intact.
+
+- [Rights of Persons with Disabilities Act, 2016 (RPD)](https://legislative.gov.in/sites/default/files/A2016-49_1.pdf)
+- [**Mandatory** Guidelines for Indian Government Websites](https://guidelines.india.gov.in/)
+
+## The current state  
+
+In the current state, blade components are not very accessible out of the box, and neither are our existing frontend products. 
+
+Here’s the screen reader accessibility of Razorpay mobile app
+
+- Missing accessibilityLabel in input box.
+- Missing accessibilityRole=button in buttons
+- Missing accessibilityRole=link in links
+- Grouping text together so they get announced at once.
+- Unlabeled password toggle icon 
+- Wrongly focus on text input's placeholder text
+
+https://user-images.githubusercontent.com/35374649/167380507-71e60b7f-d36e-42b3-b649-911400e2c66f.mp4
+
+Even though only a few of the teams are on boarded to blade, when more teams will migrate this will become a problem since every app/site will be inaccessible if we don’t implement accessibility in blade components. 
```

Comment:
```suggestion
Even though only a few teams are onboarded to blade, this could become a major problem when more teams migrate since every app/site won't be completely accessible if we don’t implement accessibility centrally in the core blade components. 
```